src/index.js

```diff
@@ -67,7 +67,14 @@ export default function factory({
     if (attachParams) req.params = params;
     let i = 0;
     const len = handlers.length;
-    const loop = async (next) => handlers[i++](req, res, next);
+    const loop = async (next) => {
+      try {
+        await Promise.resolve(handlers[i++](req, res, next)); 
```

Comment:
`Promise.resolve()` is not necessary here.

Reply:
I added the `Promise.resolve()` for the case of handlers that do not return a promise.

Reply:
It is still not needed, if `handler()` does not return a Promise, await ... will resolve immediately. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await

> If the value of the expression following the await operator is not a Promise, it's converted to a [resolved Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve).

Reply:
Ah, you are so right! Thanks for the correction.